### PR TITLE
Add Random (rectangle) erasing

### PR DESCRIPTION
**Pull request by mronta**
This is my random (rectangle) erasing proposal.

Currently this implementation does not pass the gradcheck test.
In my understanding the issue is the intrinsic non-determinism of the function, having random samplings inside it, but i could be wrong.

I would like to hear your opinions in that regard

### CODE REVIEW

**Comment by edgarriba** [comment]
@mronta thanks the PR. A couple of comments before proceeding:
1) implement the main routine in the non module function (then call it form the Module)
2) I would separate the routine in separated signatures:
  - `random_erase_rectangle` (generate random rectangle params)
  - `erase_rectangle` (given rectangle parameters, generate mask and remove)
    - this one ideally should receive the rectangle parameters so that we can backprop through it.
    - try first with float as it is but keep in mind the above comment.
  - `draw_rectangle` (a function to generate the erasing mask)
    - I open here a thread since this could be the first use case the differentiable drawing functions  that @ducha-aiki mentioned.

File: test/augmentation/test_augmentation.py [hunk 1]
```diff
@@ -623,3 +624,46 @@ def test_gradcheck(self):
...
+    @pytest.mark.parametrize("erase_scale_range", [(.001, .001), (1., 1.)])
+    @pytest.mark.parametrize("aspect_ratio_range", [(.1, .1), (10., 10.)])
+    @pytest.mark.parametrize("batch_shape", [(1, 4, 8, 15), (2, 3, 11, 7)])
+    def test_random_rectangle_erasing(
+            self, batch_shape, erase_scale_range, aspect_ratio_range):
+        input = torch.rand(batch_shape)
+        rand_rec = RandomRectangleErasing(erase_scale_range, aspect_ratio_range)
+        assert rand_rec(input).shape == batch_shape
+
+    def test_gradcheck(self):
```
Comment (ducha-aiki) [inline test/augmentation/test_augmentation.py]: Please add some toy example to check correctness of the implementation.

File: kornia/augmentation/random_erasing.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            rectangle_params[3] must be y positions tensor
+    """
+    if not (rectangle_params[0].size() == rectangle_params[1].size(
+    ) == rectangle_params[2].size() == rectangle_params[3].size()):
+        raise TypeError(
+            f"''rectangle params components must have same shape"
+        )
+
+    widths, heights, xs, ys = rectangle_params
+    mask = torch.zeros(mask_size, dtype=torch.float)
```
Comment (ducha-aiki) [inline kornia/augmentation/random_erasing.py]: It is better to do zeros(mask_size, dtype=dtype, device=device) and pass device to the function. 
Otherwise, quite significant time would be spend in CPU->GPU moving

**Comment by ducha-aiki** [comment]
One thing missing is adding entry to 
https://github.com/kornia/kornia/blob/master/docs/source/augmentation.rst

**Comment by mronta** [comment]
@ducha-aiki I also added to the docs the motion blur entries, because i did not when i committed those functions.

**Comment by ducha-aiki** [comment]
LGTM
